src/test/java/org/dependencytrack/resources/v1/BomResourceTest.java

```diff
@@ -1226,6 +1244,15 @@ public void uploadBomInvalidXmlTest() {
                   ]
                 }
                 """);
+
+        assertConditionWithTimeout(() -> kafkaMockProducer.history().size() == 1, Duration.ofSeconds(5));
```

Comment:
You can replace `assertConditionWithTimeout` with a simple `assertThat`. Events will already have been sent at the point of the assert, so no waiting necessary.

```suggestion
        assertThat(kafkaMockProducer.history()).hasSize(1);
```